python_modules/dagster/dagster/_core/definitions/utils.py

```diff
@@ -311,3 +320,89 @@ def resolve_automation_condition(
 def dedupe_object_refs(objects: Optional[Iterable[T]]) -> Sequence[T]:
     """Dedupe definitions by reference equality."""
     return list({id(obj): obj for obj in objects}.values()) if objects is not None else []
+
+
+def add_default_automation_condition_sensor(
+    sensors: Sequence["SensorDefinition"],
+    assets: Iterable[Union["AssetsDefinition", "SourceAsset"]],
+    asset_checks: Iterable["AssetsDefinition"],
+) -> Sequence["SensorDefinition"]:
+    """Adds a default automation condition sensor if the provided sensors do not already handle all
+    provided assets.
+    """
+    from dagster._core.definitions.asset_graph import AssetGraph
+    from dagster._core.definitions.automation_condition_sensor_definition import (
+        DEFAULT_AUTOMATION_CONDITION_SENSOR_NAME,
+        AutomationConditionSensorDefinition,
+    )
+
+    with disable_dagster_warnings():
+        asset_graph = AssetGraph.from_assets([*assets, *asset_checks])
+        sensor_selection = get_default_automation_condition_sensor_selection(sensors, asset_graph)
+        if sensor_selection:
+            default_sensor = AutomationConditionSensorDefinition(
+                DEFAULT_AUTOMATION_CONDITION_SENSOR_NAME, asset_selection=sensor_selection
+            )
+            sensors = [*sensors, default_sensor]
+
+    return sensors
```

Comment:
what about collision with schedules?

Reply:
only sensors can impact what asset selection we need to make here (we specifically only care about AutomationConditionSensorDefinitions)

Reply:
ah I was thrown by the docblock at the top of the function. could you reword it make it more clear?

Reply:
updated!